tests/Feature/JwtAuthorizationTest.php

```diff
@@ -0,0 +1,161 @@
+     * Perform a GET request to the test endpoint, optionally including a Bearer token in the Authorization header.
+     *
+     * @param bool $includeBearer Whether or not to include a bearer token in the request.
+     * @param Closure|null $responseHandler Response handler for endpoint. When null, a JSON containing `test` will be
+     * returned.
+     * @return TestResponse
+     */
+    private function request(bool $includeBearer, string $scope = '', ?Closure $responseHandler = null)
+    {
+        // Define a very simply testing endpoint, protected by the jwt middleware.
+        Route::middleware('jwt' . (empty($scope) ? '' : ":$scope"))
+            ->get(self::ROUTE_URI, $responseHandler ?? function () {
+                return response()->json('test');
+            });
+
+        $headers = [];
+
+        if ($includeBearer) {
+            $headers['Authorization'] = 'Bearer token';
+        }
+
+        return $this->getJson(self::ROUTE_URI, $headers);
+    }
+
+    /**
+     * Verify the request is unauthorized when no bearer token is specified.
+     */
+    public function testBearerTokenMissing()
+    {
+        $this->mockAuth0Service(null);
+
+        $this->request(false)
+            ->assertUnauthorized()
+            ->assertSee('Bearer token missing');
+    }
+
+    /**
+     * Verify the request is unauthorized when the bearer token is invalid.
+     */
+    public function testInvalidToken()
+    {
+        $this->mockAuth0Service(null);
+
+        $this->request(true)->assertUnauthorized();
+    }
+
+    /**
+     * Verify the request is forbidden when the bearer token contains no scope, but a scope is required.
+     */
+    public function testScopeRequiredNoneProvided()
+    {
+        $this->mockAuth0Service([]);
+
+        $this->request(true, 'test')
+            ->assertForbidden()
+            ->assertSee('Insufficient scope');
+    }
+
+    /**
+     * Verify the request is forbidden when the bearer token contains an empty scope, but a scope is required.
+     */
+    public function testScopeRequiredEmptyScopesProvided()
+    {
+        $this->mockAuth0Service(['scope' => '']);
+
+        $this->request(true, 'test')
+            ->assertForbidden()
+            ->assertSee('Insufficient scope');
+    }
+
+    /**
+     * Verify the request is forbidden when the bearer token only has a different scope than required.
+     */
+    public function testScopeRequiredIncorrectScopeProvided()
+    {
+        $this->mockAuth0Service(['scope' => 'nottest']);
+
+        $this->request(true, 'test')
+            ->assertForbidden()
+            ->assertSee('Insufficient scope');
+    }
+
+    /**
+     * Verify the request passes when it contains one scope only, which is required.
+     */
+    public function testScopeRequiredOneScopeProvided()
+    {
+        $this->mockAuth0Service(['scope' => 'test']);
+
+        $this->request(true, 'test')
+            ->assertOk()
+            ->assertSee('test');
+    }
+
```

Comment:
Suggestion: test no access if a scope is provided but it's not the correct one

Reply:
This should be tested by `testScopeRequiredIncorrectScopeProvided` around line 107.